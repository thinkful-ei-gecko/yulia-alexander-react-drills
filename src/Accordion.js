import React from 'react';

class Accordion extends React.Component {

  constructor(props) {
    super(props)
     this.state = {
       sectionClick: 0,
     }
  }

  static defaultProps = {
    sections: [
      {title: '', content: ''}
    ],
}

handleClick = (i) => {
  this.setState({
    sectionClick: i,
  })
}


  render() {
    console.log(this.props.sections);
    let buttons = this.props.sections.map((item, i) => {
      if (i === this.state.sectionClick) {
        return <li key={i}>
        <button key={i} onClick={() => this.handleClick(i)}>{item.title}</button>
        <p>{item.content}</p>
      </li>
      }
      return <li key={i}>
      <button key={i} onClick={() => this.handleClick(i)}>{item.title}</button>
    </li>
    })
    return (
      <ul>
        {buttons}
      </ul>
    )
  }
}

export default Accordion