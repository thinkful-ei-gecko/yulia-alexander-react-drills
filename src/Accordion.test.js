import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react';
import Accordion from './Accordion';
import sections from './STORE';

describe(`Accordion Component`, () => {

it('renders an empty li when the sections prop is not supplied', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders no sections as active by default', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
})

it('opens a clicked section', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
it('only opens one section at a time', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})