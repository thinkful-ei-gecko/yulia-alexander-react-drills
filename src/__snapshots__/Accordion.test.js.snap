// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only opens one section at a time 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion Component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders an empty li when the sections prop is not supplied 1`] = `<ul />`;

exports[`Accordion Component renders no sections as active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
